### PR TITLE
Fix Group Modal for Reservations - part II [DDFLSBP-408]

### PR DESCRIPTION
**Pull request by clausbruun**
#### Link to issue

https://reload.atlassian.net/browse/DDFLSBP-408

#### Description

- General cleanup of the group reservations modal.
- In cases where the expected redeem date is unknown, it will now display 'You are in the reservation queue.'
- The link 'See details for the material' has been removed.
- The design of the expected redeem date has been adjusted in accordance with Figma.
- Alignment has been verified.

#### Screenshot of the result

image

image

image

#### Additional comments or questions

*

### CODE REVIEW

File: src/components/GroupModal/GroupModalReservationsList.tsx [hunk 1]
```diff
@@ -75,45 +76,57 @@ const GroupModalReservationsList: FC<GroupModalReservationsListProps> = ({
...
+            });
+          }
+
+          const statusBadgeComponent = statusText ? (
+            <StatusBadge
+              badgeDate={expiryDate}
+              neutralText={statusText}
+              infoText=""
+            />
+          ) : null;
```
Comment (spaceo) [inline src/components/GroupModal/GroupModalReservationsList.tsx:112]: I think this should be put into a helper function in a `helper.ts` file inside the `GroupModal` folder.
Personally I am not a big fan of `let` but thats maybe more a matter of taste.
Here is a suggestion for a typed helper function:
```typescript 
export const getStatusText = ({
  identifier,
  state,
  pickupDeadline,
  faust,
  numberInQueue,
  t
}: ReservationType & { t: UseTextFunction }) => {
  if (identifier && state === "reserved") {
    if (pickupDeadline === null || pickupDeadline === "") {
      return t("reservationListYouAreInQueueText");
    }

    return t("reservationListAvailableInText", {
      placeholders: {
        "@count": daysBetweenTodayAndDate(pickupDeadline ?? "")
      }
    });
  }

  if (faust && numberInQueue) {
    return t("dashboardNumberInLineText", {
      count: numberInQueue,
      placeholders: { "@count": numberInQueue }
    });
  }

  return "";
};
```

File: src/components/GroupModal/GroupModalReservationsList.tsx [hunk 2]
```diff
@@ -75,45 +76,57 @@ const GroupModalReservationsList: FC<GroupModalReservationsListProps> = ({
...
+                openDetailsModal={openDetailsModal}
+                key={reservationId(material)}
+                selected={selected}
+                onMaterialChecked={onMaterialChecked}
+                disabled={false}
+                statusMessageComponentMobile={null}
+                statusMessageComponentDesktop={null}
+              />
+            )
           );
```
Comment (spaceo) [inline src/components/GroupModal/GroupModalReservationsList.tsx:130]: Nice to get it more tight/simple :)

**Review by spaceo — COMMENTED** [review]
Looks good!
I have one comment/suggestion

**Comment by clausbruun** [comment]
> Looks good! I have one comment/suggestion

@spaceo Nice suggestion! When you have a moment, could you please take another look at the revised PR?

**Review by spaceo — APPROVED** [review]
LGTM 👍

**Pull request merged**